tests/util/test_languages.py

```diff
@@ -82,3 +84,90 @@ def test_name_for_languageset(self):
         eq_(u"español/English/Chinese", m(["spa","eng","chi"]))
         assert_raises(ValueError(m, ["eng, nxx"]))
+
+
+class TestLanguageNames(object):
+    """Test our (very rough) ability to map from natural-language names
+    of languages to ISO-639-2 language codes.
+    """
+
+    def test_name_to_codes(self):
+        # Verify that the name_to_codes dictionary was populated
+        # appropriately.
+        d = LanguageNames.name_to_codes
+
+        def coded(name, code):
+            # In almost all cases, a human-readable language name maps to
+            # a set containing a single ISO-639-2 language code.
+            eq_(set([code]), d[name])
+
+        # English-language names work.
+        coded("english", "eng")
+        coded("french", "fre")
+        coded("irish", "gle")
+        coded("tokelau", "tkl")
+        coded("persian", "per")
+
+        # (Some) native-language names work
+        coded("francais", "fre")
+        coded("espanol", "spa")
+        coded("castellano", "spa")
+        for item in LanguageCodes.NATIVE_NAMES_RAW_DATA:
+            coded(item['nativeName'].lower(),
+                  LanguageCodes.two_to_three[item['code']])
+
+        # Languages associated with a historical period are not mapped
+        # to codes.
+        eq_(set(), d['irish, old (to 900)'])
+
+        # This general rule would exclude Greek ("Greek, Modern
+        # (1453-)") and Occitan ("Occitan (post 1500)"), so we added
+        # them manually.
+        coded('greek', 'gre')
+        coded('occitan', 'oci')
+
+        # Languages associated with a geographical area, such as "Luo
+        # (Kenya and Tanzania)", can be looked up without that area.
+        coded('luo', 'luo')
+
+        # This causes a little problem for Tonga: there are two
+        # unrelated languages called 'Tonga', and the geographic area
+        # is the only way to distinguish them. For now, we map 'tonga'
+        # to both ISO codes. (This is why name_to_codes is called that
+        # rather than name_to_code.)
+        eq_(set(['ton', 'tog']), d['tonga'])
+
+        # Language families such as "Himacahli languages" can be
+        # looked up without the " languages".
+        coded('himachali', 'him')
+
+        # Language groups such as "Bantu (Other)" can be looked up
+        # without the "(Other)".
+        coded('south american indian', 'sai')
+        coded('bantu', 'bnt')
+
+        # If a language is known by multiple English names, lookup on
+        # any of those names will work.
+        for i in "Blissymbols; Blissymbolics; Bliss".split(";"):
+            coded(i.strip().lower(), 'zbl')
+
+    def test_name_re(self):
+        # Verify our ability to find language names inside text.
+        def find(text, expect):
+            match = LanguageNames.name_re.search(text)
+            if not match:
+                return match
+            return match.groups()[0]
+
+            eq_(expect, LanguageNames.find_language(text))
```

Comment:
I don't see a reference to `find_language` and this probably doesn't run because it's after the `return`.

Reply:
Yeah, this is old code.